### ISSUE TITLE
Some code not transpiled with @vue/cli-service 4.4.6

### ISSUE THREAD

**Issue by rndmerle** [vuejs/vue-cli#5617]
### Version
4.4.6

### Environment info
```
  System:
    OS: macOS 10.15.5
    CPU: (8) x64 Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz
  Binaries:
    Node: 12.16.3 - /usr/local/opt/node@12/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/opt/node@12/bin/npm
  Browsers:
    Chrome: 83.0.4103.116
    Edge: Not Found
    Firefox: 76.0
    Safari: 13.1.1
  npmPackages:
    @egoist/vue-to-react:  1.1.0
    @storybook/vue: ^5.3.19 => 5.3.19
    @tinymce/tinymce-vue: ^3.2.2 => 3.2.2
    @vue/babel-helper-vue-jsx-merge-props:  1.0.0
    @vue/babel-plugin-transform-vue-jsx:  1.1.2
    @vue/babel-preset-app:  4.4.6
    @vue/babel-preset-jsx:  1.1.2
    @vue/babel-sugar-functional-vue:  1.1.2
    @vue/babel-sugar-inject-h:  1.1.2
    @vue/babel-sugar-v-model:  1.1.2
    @vue/babel-sugar-v-on:  1.1.2
    @vue/cli-overlay:  4.4.6
    @vue/cli-plugin-babel: 4.4.6 => 4.4.6
    @vue/cli-plugin-router: 4.4.6 => 4.4.6
    @vue/cli-plugin-typescript: 4.4.6 => 4.4.6
    @vue/cli-plugin-unit-jest: 4.4.6 => 4.4.6
    @vue/cli-plugin-vuex: 4.4.6 => 4.4.6
    @vue/cli-service: 4.4.6 => 4.4.6
    @vue/cli-shared-utils:  4.4.6
    @vue/component-compiler-utils:  3.1.2 (3.1.1)
    @vue/composition-api: 0.6.1 => 0.6.1
    @vue/eslint-config-prettier: ^6.0.0 => 6.0.0
    @vue/preload-webpack-plugin:  1.1.1
    @vue/test-utils: ^1.0.0-beta.28 => 1.0.0-beta.28
    @vue/web-component-wrapper:  1.2.0
    eslint-plugin-vue: ^6.2.2 => 6.2.2
    jest-serializer-vue:  2.0.2
    typescript: 3.9.5 => 3.9.5 (3.9.3)
    vue: ^2.6.11 => 2.6.11
    vue-apexcharts: ^1.5.3 => 1.5.3
    vue-clamp: ^0.3.0 => 0.3.0
    vue-class-component: ^7.2.3 => 7.2.3
    vue-cli-plugin-storybook: ^1.2.2 => 1.2.2
    vue-docgen-api:  4.8.1
    vue-docgen-loader:  1.4.0
    vue-eslint-parser: ^7.1.0 => 7.1.0 (7.0.0)
    vue-hot-reload-api:  2.3.4
    vue-jest: ^3.0.5 => 3.0.5
    vue-loader:  15.9.2
    vue-plyr: ^6.0.4 => 6.0.4
    vue-property-decorator: 8.5.0 => 8.5.0
    vue-router: 3.3.4 => 3.3.4
    vue-selectable: ^0.5.0 => 0.5.0
    vue-server-renderer: ^2.6.11 => 2.6.11
    vue-style-loader:  4.1.2
    vue-template-compiler: ^2.6.11 => 2.6.11
    vue-template-es2015-compiler:  1.9.1
    vue-text-mask: ^6.1.2 => 6.1.2
    vue-typed-mixins: ^0.2.0 => 0.2.0
    vue-upload-component: ^2.8.19 => 2.8.20
    vuedraggable: ^2.23.2 => 2.23.2
    vuex: ^3.4.0 => 3.4.0
  npmGlobalPackages:
    @vue/cli: 4.3.1
```

### Steps to reproduce
I upgraded vue-cli packages from 4.4.4 to 4.4.6.
I built the app with --modern parameter

### What is expected?
App still working on IE11

### What is actually happening?
The app fails on IE11 because some code is not transpiled to ES5 (there are some const and fat arrow functions still, in my own code, not lib's).

---
If I just switch back @vue/cli-service to 4.4.5 it works again.

If you've no idea what could cause this in the recent commits I'll try to set up a minimal reproduction.

**Comment by sodatea** [comment]
Could you provide a reproduction repo? It's very unlikely that a Vue CLI update broke your project, because all the changes shipped in the v4.4.6 are for the CLI users, none of them affects the packages installed as dependencies in the project.

So it must be some other packages upgraded at the same time.

**Comment by rndmerle** [comment]
I still don't understand how but it's linked to an other dependency and not directly to vue/cli.
I'll close the issue, sorry for your time.
